core/src/main/scala/akka/persistence/cassandra/cleanup/Cleanup.scala

```diff
@@ -0,0 +1,132 @@
+ * of the list. It's possible to parallelize the deletes by running several cleanup operations
+ * at the same time operating on different sets of `persistenceIds`.
+ */
+@ApiMayChange
+final class Cleanup(system: ActorSystem, settings: CleanupSettings) {
+
+  def this(system: ActorSystem) =
+    this(system, new CleanupSettings(system.settings.config.getConfig("akka.persistence.cassandra.cleanup")))
+
+  import settings._
+  import system.dispatcher
+
+  private val log = Logging(system, getClass)
+  private val journal = Persistence(system).journalFor(journalPlugin)
+  private val snapshotStore =
+    if (snapshotPlugin != "" || system.settings.config.getString("akka.persistence.snapshot-store.plugin").nonEmpty)
+      Some(Persistence(system).snapshotStoreFor(snapshotPlugin))
+    else None
+  private implicit val askTimeout: Timeout = operationTimeout
+
+  /**
+   * Delete everything related to the given list of `persistenceIds`. All events and snapshots are deleted.
+   */
+  def deleteAll(persistenceIds: immutable.Seq[String], neverUsePersistenceIdAgain: Boolean): Future[Done] = {
+    foreach(persistenceIds, "deleteAll", pid => deleteAll(pid, neverUsePersistenceIdAgain))
+  }
+
+  /**
+   * Delete everything related to one single `persistenceId`. All events and snapshots are deleted.
+   */
+  def deleteAll(persistenceId: String, neverUsePersistenceIdAgain: Boolean): Future[Done] = {
+    deleteAllEvents(persistenceId, neverUsePersistenceIdAgain).flatMap(_ => deleteAllSnapshots(persistenceId))
+  }
+
+  /**
+   * Delete all events related to the given list of `persistenceIds`. Snapshots are not deleted.
+   */
+  def deleteAllEvents(persistenceIds: immutable.Seq[String], neverUsePersistenceIdAgain: Boolean): Future[Done] = {
+    foreach(persistenceIds, "deleteAllEvents", pid => deleteAllEvents(pid, neverUsePersistenceIdAgain))
+  }
+
+  /**
+   * Delete all events related to one single `persistenceId`. Snapshots are not deleted.
+   */
+  def deleteAllEvents(persistenceId: String, neverUsePersistenceIdAgain: Boolean): Future[Done] = {
+    (journal ? CassandraJournal.DeleteAllEvents(persistenceId, neverUsePersistenceIdAgain)).mapTo[Done]
+  }
+
+  /**
+   * Delete all snapshots related to the given list of `persistenceIds`. Events are not deleted.
+   */
+  def deleteAllSnapshots(persistenceIds: immutable.Seq[String]): Future[Done] = {
+    if (snapshotStore.isDefined)
+      foreach(persistenceIds, "deleteAllSnapshots", pid => deleteAllSnapshots(pid))
+    else
+      Future.successful(Done)
+  }
+
+  /**
+   * Delete all snapshots related to one single `persistenceId`. Events are not deleted.
+   */
+  def deleteAllSnapshots(persistenceId: String): Future[Done] = {
+    snapshotStore match {
+      case Some(snapshotStoreRef) =>
+        (snapshotStoreRef ? CassandraSnapshotStore.DeleteAllsnapshots(persistenceId)).mapTo[Done]
+      case None => Future.successful(Done)
+    }
+  }
+
+  private def foreach(
```

Comment:
it can be parallelized by kicking off several batches of persistenceIds. keeping each run sequential. I was thinking that if something fails it is good to know how far it got in the list of persistenceIds, and that would be more uncertain if not running it sequentially